lib/erb_lint/cli.rb

```diff
@@ -95,13 +112,47 @@ def run(args = ARGV)
 
     private
 
+    attr_reader :cache, :config
+
+    def run_on_file(runner, filename)
+      file_content = read_content(filename)
+      if with_cache? && !autocorrect?
+        run_using_cache(runner, filename, file_content)
+      else
+        file_content = run_with_corrections(runner, filename, file_content)
+      end
+      log_offense_stats(runner, filename)
+      file_content
+    end
+
+    def run_using_cache(runner, filename, file_content)
+      if cache.include?(filename) && !autocorrect?
+        runner.restore_offenses(cache.get(filename, file_content))
+        cache.add_hit(filename) if prune_cache?
+      else
+        run_with_corrections(runner, filename, file_content)
+        cache[filename] = runner.offenses.map(&:to_json_format).to_json
+        cache.add_new_result(filename) if prune_cache?
+      end
+    end
+
     def autocorrect?
       @options[:autocorrect]
     end
 
-    def run_with_corrections(runner, filename)
-      file_content = read_content(filename)
+    def with_cache?
+      @options[:with_cache]
+    end
 
+    def prune_cache?
+      @options[:prune_cache]
+    end
```

Comment:
I wonder if we can also be a bit more smart here? Something what Rubocop does with max files in cache and then we prune the oldest when it's getting higher.

Reply:
We definitely could but, naively, you can just run `erblint <files> --with-cache --prune-cache` every time and the code as written will remove outdated cache files only. Maybe we can leave more smart cache pruning for a future PR?